commands/addResource.ts

```diff
@@ -1,56 +1,270 @@
-                    break;
-                default:
-                    return new Error('something bad happened')
-            }
-        })
+  let contributor;
+  if (!await isContributor(interaction.user)) {
+    await interaction.reply(`it looks like you are not a contributor yet!\nPlease add yourself using: ${inlineCode('/add-contributor')}`)
+    return
+  }
+  else {
+    const contributorResponse = await findContributor(interaction.user);
+    if (contributorResponse) {
+      contributor = contributorResponse.getId();
+    }
+  }
+
+  const url = interaction.options.getString('url');
+  if (!url) {
+    interaction.reply({ content: 'URL required, please try submitting again.', ephemeral: true });
+    return;
+  }
+
+  if (!isValidUrl(url)) {
+    interaction.reply({ content: 'Invalid URL provided, please check it before submitting again.', ephemeral: true });
+    return;
+  }
+
+  const foundResouce = await findResourceByUrl(url);
+  if (foundResouce) {
+    interaction.reply({ content: 'A resource with this URL already exists.', ephemeral: true });
+    return;
+  }
+
+  await interaction.deferReply();
+
+  const resource = getSanitizedResourceInfo(interaction);
+  resource.contributor = contributor;
+
+  const resourceEmbed = buildEmbed(resource);
+  const tags = await readTags();
+  const tagsOptions = tags.map(tag => ({ label: tag.name, value: tag.id }));
+  const tagsRow = new MessageActionRow().addComponents(
+    new MessageSelectMenu()
+      .setCustomId('tags')
+      .setPlaceholder('Select tags')
+      .setMaxValues(Math.min(tagsOptions.length, 25))
+      .addOptions(tagsOptions),
+  );
+
+  const blockchain = await readBlockchain();
+  const bcOptions = blockchain.map(bc => ({ label: bc.name, value: bc.id }));
+  const bcRow = new MessageActionRow().addComponents(
+    new MessageSelectMenu()
+      .setCustomId('blockchain')
+      .setPlaceholder('Select blockchain')
+      .setMaxValues(Math.min(bcOptions.length, 25))
+      .addOptions(bcOptions),
+  );
+
+  const categories = await readCategory();
+  const categoryOptions = categories.map(category => ({ label: category.name, value: category.id }));
+  const categoryRow = new MessageActionRow().addComponents(
+    new MessageSelectMenu()
+      .setCustomId('category')
+      .setPlaceholder('Select categories')
+      .setMaxValues(Math.min(categoryOptions.length, 25))
+      .addOptions(categoryOptions),
+  );
+
+  const authors = await readAuthors();
+  const authorOptions = authors.map(author => ({ label: author.name, value: author.id }));
+  const authorRow = new MessageActionRow().addComponents(
+    new MessageSelectMenu()
+      .setCustomId('author')
+      .setPlaceholder('Select author')
+      .addOptions(authorOptions),
+  );
+
+  const selectionRows = [authorRow, bcRow, categoryRow, tagsRow];
+
+  await interaction.editReply({
+    embeds: [resourceEmbed],
+    content: 'Tell me about the resource',
+    components: selectionRows,
+  });
```

Comment:
Could we make this ephemeral too?

Reply:
When calling `editReply` `ephemeral` isn't a valid option. I think that I missed it off the new `deferred` response. Will look into it.